contracts/ButtonToken.sol

```diff
@@ -172,12 +172,12 @@ contract ButtonToken is IButtonToken, Initializable, OwnableUpgradeable {
     function updateOracle(address oracle_) public override onlyOwner {
         uint256 price;
         bool valid;
-        (price, valid) = IOracle(oracle_).getData();
-        require(valid, "ButtonToken: unable to fetch data from oracle");
 
         oracle = oracle_;
```

Comment:
i think it has to be before because we want to check that the queried price is valid from the new oracle -- `_queryPrice()` just uses the oracle from state

It also fails on initialization that way because `_queryPrice()` tries to query oracle 0x0